### PR TITLE
update modules version without consensus break

### PR DESCRIPTION
**Pull request by Galadrin**
update to latest go module
upgrade to go 1.18

node sync well

### CODE REVIEW

**Comment by Galadrin** [comment]
```
$ make test
ok  	github.com/NXTPOP/teritori-chain/app	(cached)
?   	github.com/NXTPOP/teritori-chain/app/helpers	[no test files]
?   	github.com/NXTPOP/teritori-chain/app/params	[no test files]
?   	github.com/NXTPOP/teritori-chain/cmd/teritorid	[no test files]
ok  	github.com/NXTPOP/teritori-chain/cmd/teritorid/cmd	(cached)
?   	github.com/NXTPOP/teritori-chain/x/airdrop	[no test files]
?   	github.com/NXTPOP/teritori-chain/x/airdrop/client/cli	[no test files]
ok  	github.com/NXTPOP/teritori-chain/x/airdrop/keeper	(cached)
ok  	github.com/NXTPOP/teritori-chain/x/airdrop/types	(cached)
?   	github.com/NXTPOP/teritori-chain/x/nftstaking	[no test files]
?   	github.com/NXTPOP/teritori-chain/x/nftstaking/client/cli	[no test files]
?   	github.com/NXTPOP/teritori-chain/x/nftstaking/keeper	[no test files]
?   	github.com/NXTPOP/teritori-chain/x/nftstaking/types	[no test files]
```

File: go.mod [hunk 1]
```diff
@@ -140,24 +142,24 @@ require (
...
 
 require (
-	github.com/cosmos/ibc-go/v2 v2.0.0
 	github.com/hdevalence/ed25519consensus v0.0.0-20220222234857-c00d1f31bab3 // indirect
 	github.com/strangelove-ventures/packet-forward-middleware/v2 v2.1.1
 )
 
 replace (
 	github.com/cosmos/iavl => github.com/cosmos/iavl v0.17.3
+	github.com/cosmos/ledger-go => github.com/empowerchain/ledger-go v0.9.3-0.20220531092147-dab46ea76974
```
Comment (n0izn0iz) [inline go.mod:162]: please add a comment explaining why the replace is done
I would prefer that we stick to upstream but this seems ok

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (n0izn0iz) [reply go.mod:162]: Don't merge this, I tried to replay the chain and I have a consensus error at height 8487
```
7:48PM INF indexed block height=8487 module=txindex
panic: Failed to process committed block (8488:E5B65F246B14B9EA8DE826780811BCF02263FC047129F17AE50FDD599D9745AB): wrong Block.Header.LastResultsHash.  Expected E8F73548DE5F44CA3BCB0B8EB69021C62B3EA99F5A752F7C1F87BC522D8CDD99, got EA1C6DB1ED5A6472AAEC3481CCDFB6F8DA2201BF22E70467CAB05342C6887B20

goroutine 45 [running]:
github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).poolRoutine(0x14000599c00, 0x0)
	github.com/tendermint/tendermint@v0.34.19/blockchain/v0/reactor.go:401 +0xed8
created by github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).OnStart
	github.com/tendermint/tendermint@v0.34.19/blockchain/v0/reactor.go:110 +0x80
```

Comment (Galadrin) [reply go.mod:162]: > please add a comment explaining why the replace is done I would prefer that we stick to upstream but this seems ok

This change allow to use the new Ledger Nano S plus.
the pull request is not merged in upstream

Comment (Galadrin) [reply go.mod:162]: > Don't merge this, I tried to replay the chain and I have an error at height 8487

have you sync the master branch or the teritory-testnet-v2 branch with the patch ?

my patch is on the master branch.

I've restart a new node sync without issue (height >20k now) and will declare a validator with it to have the consenssus module enable.

```
$ ~/go/bin/teritorid version --long
name: teritori
server_name: teritorid
version: teritori-testnet-v2-e4f8e712f6e0f83c08f2888687a17a3d0b4ef12b
commit: e4f8e712f6e0f83c08f2888687a17a3d0b4ef12b
build_tags: netgo ledger,
go: go version go1.18.2 linux/amd64
build_deps:
- contrib.go.opencensus.io/exporter/stackdriver@v0.13.4
- filippo.io/edwards25519@v1.0.0-rc.1
- github.com/99designs/keyring@v1.1.6
- github.com/ChainSafe/go-schnorrkel@v0.0.0-20200405005733-88cbf1b4c40d
- github.com/CosmWasm/wasmd@v0.27.0
- github.com/CosmWasm/wasmvm@v1.0.0
- github.com/Workiva/go-datastructures@v1.0.53
- github.com/armon/go-metrics@v0.3.10
- github.com/aybabtme/rgbterm@v0.0.0-20170906152045-cc83f3b3ce59
- github.com/beorn7/perks@v1.0.1
- github.com/bgentry/speakeasy@v0.1.0
- github.com/blendle/zapdriver@v1.3.1
- github.com/btcsuite/btcd@v0.22.0-beta
- github.com/cespare/xxhash/v2@v2.1.2
- github.com/coinbase/rosetta-sdk-go@v0.7.0
- github.com/confio/ics23/go@v0.7.0
- github.com/cosmos/btcutil@v1.0.4
- github.com/cosmos/cosmos-sdk@v0.45.6
- github.com/cosmos/go-bip39@v1.0.0
- github.com/cosmos/iavl@v0.17.3 => github.com/cosmos/iavl@v0.17.3
- github.com/cosmos/ibc-go/v3@v3.0.1
- github.com/cosmos/ledger-cosmos-go@v0.11.1
- github.com/cosmos/ledger-go@v0.9.2 => github.com/empowerchain/ledger-go@v0.9.3-0.20220531092147-dab46ea76974
- github.com/davecgh/go-spew@v1.1.1
- github.com/desertbit/timer@v0.0.0-20180107155436-c41aec40b27f
- github.com/dfuse-io/logging@v0.0.0-20210109005628-b97a57253f70
- github.com/dvsekhvalnov/jose2go@v0.0.0-20200901110807-248326c1351b
- github.com/ethereum/go-ethereum@v1.10.20
- github.com/fatih/color@v1.13.0
- github.com/felixge/httpsnoop@v1.0.1
- github.com/fsnotify/fsnotify@v1.5.4
- github.com/gagliardetto/binary@v0.6.1
- github.com/gagliardetto/solana-go@v1.2.1
- github.com/gagliardetto/treeout@v0.1.4
- github.com/go-kit/kit@v0.12.0
- github.com/go-kit/log@v0.2.0
- github.com/go-logfmt/logfmt@v0.5.1
- github.com/godbus/dbus@v0.0.0-20190726142602-4481cbc300e2
- github.com/gogo/gateway@v1.1.0
- github.com/gogo/protobuf@v1.3.3 => github.com/regen-network/protobuf@v1.3.3-alpha.regen.1
- github.com/golang/groupcache@v0.0.0-20210331224755-41bb18bfe9da
- github.com/golang/protobuf@v1.5.2
- github.com/golang/snappy@v0.0.4
- github.com/google/btree@v1.0.0
- github.com/google/gofuzz@v1.2.0
- github.com/google/orderedcode@v0.0.1
- github.com/gorilla/handlers@v1.5.1
- github.com/gorilla/mux@v1.8.0
- github.com/gorilla/websocket@v1.5.0
- github.com/gravity-devs/liquidity@v1.5.1
- github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0
- github.com/grpc-ecosystem/grpc-gateway@v1.16.0
- github.com/grpc-ecosystem/grpc-gateway/v2@v2.0.1
- github.com/gsterjov/go-libsecret@v0.0.0-20161001094733-a6f4afe4910c
- github.com/gtank/merlin@v0.1.1
- github.com/gtank/ristretto255@v0.1.2
- github.com/hashicorp/go-immutable-radix@v1.3.1
- github.com/hashicorp/golang-lru@v0.5.5-0.20210104140557-80c98217689d
- github.com/hashicorp/hcl@v1.0.0
- github.com/hdevalence/ed25519consensus@v0.0.0-20220222234857-c00d1f31bab3
- github.com/improbable-eng/grpc-web@v0.14.1
- github.com/json-iterator/go@v1.1.12
- github.com/klauspost/compress@v1.13.6
- github.com/lib/pq@v1.10.4
- github.com/libp2p/go-buffer-pool@v0.0.2
- github.com/logrusorgru/aurora@v2.0.3+incompatible
- github.com/magiconair/properties@v1.8.6
- github.com/mattn/go-colorable@v0.1.12
- github.com/mattn/go-isatty@v0.0.14
- github.com/matttproud/golang_protobuf_extensions@v1.0.1
- github.com/mimoo/StrobeGo@v0.0.0-20181016162300-f8f6d4d2b643
- github.com/minio/highwayhash@v1.0.2
- github.com/mitchellh/go-homedir@v1.1.0
- github.com/mitchellh/go-testing-interface@v1.14.1
- github.com/mitchellh/mapstructure@v1.4.3
- github.com/modern-go/concurrent@v0.0.0-20180306012644-bacd9c7ef1dd
- github.com/modern-go/reflect2@v1.0.2
- github.com/mostynb/zstdpool-freelist@v0.0.0-20201229113212-927304c0c3b1
- github.com/mr-tron/base58@v1.2.0
- github.com/mtibben/percent@v0.2.1
- github.com/pelletier/go-toml@v1.9.4
- github.com/pkg/errors@v0.9.1
- github.com/pmezard/go-difflib@v1.0.0
- github.com/prometheus/client_golang@v1.12.2
- github.com/prometheus/client_model@v0.2.0
- github.com/prometheus/common@v0.32.1
- github.com/prometheus/procfs@v0.7.3
- github.com/rakyll/statik@v0.1.7
- github.com/rcrowley/go-metrics@v0.0.0-20200313005456-10cdbea86bc0
- github.com/regen-network/cosmos-proto@v0.3.1
- github.com/rs/cors@v1.8.2
- github.com/rs/zerolog@v1.26.0
- github.com/spf13/afero@v1.8.2
- github.com/spf13/cast@v1.5.0
- github.com/spf13/cobra@v1.5.0
- github.com/spf13/jwalterweatherman@v1.1.0
- github.com/spf13/pflag@v1.0.5
- github.com/spf13/viper@v1.11.0
- github.com/strangelove-ventures/packet-forward-middleware/v2@v2.1.1
- github.com/stretchr/testify@v1.7.2
- github.com/subosito/gotenv@v1.2.0
- github.com/syndtr/goleveldb@v1.0.1-0.20220614013038-64ee5596c38a
- github.com/tendermint/btcd@v0.1.1
- github.com/tendermint/crypto@v0.0.0-20191022145703-50d29ede1e15
- github.com/tendermint/go-amino@v0.16.0
- github.com/tendermint/tendermint@v0.34.19
- github.com/tendermint/tm-db@v0.6.7
- github.com/teris-io/shortid@v0.0.0-20201117134242-e59966efd125
- github.com/tidwall/gjson@v1.9.3
- github.com/tidwall/match@v1.1.1
- github.com/tidwall/pretty@v1.2.0
- github.com/zondax/hid@v0.9.0
- go.opencensus.io@v0.23.0
- go.uber.org/atomic@v1.9.0
- go.uber.org/multierr@v1.7.0
- go.uber.org/zap@v1.19.1
- golang.org/x/crypto@v0.0.0-20220411220226-7b82a4e95df4
- golang.org/x/net@v0.0.0-20220607020251-c690dde0001d
- golang.org/x/sys@v0.0.0-20220520151302-bc2c85ada10a
- golang.org/x/term@v0.0.0-20210927222741-03fcf44c2211
- golang.org/x/text@v0.3.7
- google.golang.org/genproto@v0.0.0-20220714211235-042d03aeabc9
- google.golang.org/grpc@v1.48.0 => google.golang.org/grpc@v1.33.2
- google.golang.org/protobuf@v1.28.0
- gopkg.in/ini.v1@v1.66.4
- gopkg.in/yaml.v2@v2.4.0
- gopkg.in/yaml.v3@v3.0.1
- nhooyr.io/websocket@v1.8.6
cosmos_sdk_version: v0.45.6

```

**Comment by gopherjk121** [comment]
This is the PR into master and our testnet v2 is running on teritori-testnet-v2 branch. IT's safe to merge this into master.

File: go.mod (hunk 1 above)
Comment (n0izn0iz) [reply go.mod:162]: > > please add a comment explaining why the replace is done I would prefer that we stick to upstream but this seems ok
> 
> This change allow to use the new Ledger Nano S plus. the pull request is not merged in upstream

I saw that it's for the nano s plus, I wanted a comment in the go.mod so we don't have to dig the repo to understand

I synced on your branch, so master, but the only changes besides yours are README stuff so it's the same, no?

**Pull request merged**